### ISSUE TITLE
deep-rnn for har

### ISSUE THREAD

**Issue by Drug1996** [Drug1996/DeepNAR#7]
### Journals
**Sensors** (2017)
[17_deep_recurrent_neural_network_for_har.pdf](https://github.com/Drug1996/DeepNAR/files/3316502/17_deep_recurrent_neural_network_for_har.pdf)
### Author and affiliation
**Author**: _Abdulmajid Murad and Jae-Young Pyun_
**Affiliation**: _Department of Information Communication Engineering, Chosun University, 375 Susuk-dong, Dong-gu, Gwangju 501-759, Korea_
### Motivation
Typical models (typical machine learning and CNN) are unadaptable to a wide range of activity-recognition configurations and require fixed-length input windows.
### Database
Public database: **UCI-HAD, USC-HAD, Opportunity, Daphnet FOG, Skoda**
![image](https://user-images.githubusercontent.com/32189927/59958485-76503b00-94e2-11e9-837e-92932bb370a8.png)

### Method
- Three proposed structures: **_Unidirectional DRNNs_, _Bidirectional DRNNs_, _Cascaded unidirectional and bidirectional DRNNs_**
- Dropout between the layesrs
- Used late-fusion technique in which the classification decision are combined for the overall prediction of a window;

<div align=center><img src="https://user-images.githubusercontent.com/32189927/59958255-7cdcb380-94de-11e9-9844-632cb8ee1441.png" width="50%" height="50%"></div>
<div align=center><img src="https://user-images.githubusercontent.com/32189927/59958391-00979f80-94e1-11e9-80af-ce7464cc757c.png" width="50%" height="50%"><p>Unidirectional LSTM-based DRNNs</div>
<div align=center><img src="https://user-images.githubusercontent.com/32189927/59958430-96cbc580-94e1-11e9-9bfa-278a238d7117.png" width="50%" height="50%"><p>Bidirectional LSTM-based DRNNs</div>
<div align=center><img src="https://user-images.githubusercontent.com/32189927/59958442-c2e74680-94e1-11e9-8d73-344d6fc9691e.png" width="50%" height="50%"><p>Cascaded unidirectional and bidirectional LSTM-based DRNNs</div>

### Contribution
They presented three LSTM-based DRNN architectures for HAR, which outperform other state-of-the-art methods by testing on four benchmark datasets.
<div align=center><img src="https://user-images.githubusercontent.com/32189927/59959233-195a8200-94ee-11e9-96dd-b0da35651787.png" width="70%" height="70%"><p>Results of the proposed model for the Opportunity dataset</div>

### Insight
- When training the model, there is no valid set. The model and the result cannot be trusted, which may be overfitted to the test set;
- Borrow Deep-RNNs from other domain and test on HAR;
### Useful information
The optimal window length of a dataset depends on the sampling rate and the type of activities performed;
